src/components/visualizations/chart/chartOptionsBuilder.ts

```diff
@@ -721,13 +721,38 @@ export function getSeries(
 
 export const customEscape = (str: string) => str && escape(unescape(str));
 
-const renderTooltipHTML = (textData: string[][]): string => {
-    return `<table class="tt-values gd-viz-tooltip-table">${textData
+const renderTooltipHTML = (textData: string[][], chartType?: string, chartWidth?: number): string => {
+    let tableStyle = "";
+    let columnTitleStyle = "";
+    let columnValueStyle = "";
+
+    if (chartType && chartWidth && chartWidth < 340) {
+        // https://github.com/highcharts/highcharts/blob/master/ts/parts/Chart.ts#L1863
+        // plotLeft = 10, marginRight = 10;
+        const chartPadding = chartType !== VisualizationTypes.TREEMAP ? 0 : 20;
+
+        // .gd-viz-tooltip-content
+        // padding: 13px 13px 9px;
+        // border: 1px solid transparentize($bubble-light-border-base, 0.3);
+        const tableMaxWidth = chartWidth - 28 - chartPadding; // 28 = 1 + 13 + 13 + 1
+        const columnMaxWidth = tableMaxWidth / 2; // 2 columns
+
+        tableStyle = ` style="max-width: ${tableMaxWidth}px;-webkit-border-horizontal-spacing: 0;"`;
+        columnTitleStyle = ` style="max-width: ${columnMaxWidth}px;"`;
+        // .gd-viz-tooltip-table-value => padding: 0 0 0 5px;
+        columnValueStyle = ` style="max-width: ${columnMaxWidth - 5}px;"`;
+    }
```

Comment:
I think we can create new `/gooddata-react-components/src/constants/tooltip.ts` then use it here
```
export const MAX_CHART_WIDTH = 340;

// https://github.com/highcharts/highcharts/blob/master/ts/parts/Chart.ts#
// plotLeft = 10, marginRight = 10;
export const CHART_PADDING = 20;

// border: 1px solid transparentize($bubble-light-border-base, 0.3);
const TOOLTIP_BORDER = 1;

// .gd-viz-tooltip-content
// padding: 13px 13px 9px;
const TOOLTIP_CONTENT_PADDING = 13;

export const TOOLTIP_PADDING = TOOLTIP_BORDER + TOOLTIP_CONTENT_PADDING + TOOLTIP_CONTENT_PADDING + TOOLTIP_BORDER; // 28

export const TOOLTIP_COLUMN_NUMBER = 2;

// .gd-viz-tooltip-table-value => padding: 0 0 0 5px;
export const TOOLTIP_VALUE_PADDING = 5;
```

Reply:
updated :)